modules/commons/src/main/java/org/apache/synapse/commons/staxon/core/event/SimpleXMLEventAllocator.java

```diff
@@ -378,8 +378,10 @@ public QName getName() {
         }
 
         @Override
-        public Iterator<?> getNamespaces() {
-            return namespaces.iterator();
+        public Iterator<Namespace> getNamespaces() {
+            @SuppressWarnings("unchecked")
```

Comment:
What's the reason for SuppressWarnings?

Reply:
This is due to the casting of a List type. The casting is Upcast. 
```List<Namespace> namespaceList = (List<Namespace>)(List<?>) namespaces;```
Otherwise need to iterate through the whole list and copy it. Or change a considerable amount of Method Signatures and their usage. Btw, do we have an easy way to get rid of this warning?